### PR TITLE
Show version info in console

### PR DESCRIPTION
**Pull request by nmattia**
This bakes the commit sha (if possible) and shows it in the browser
console for easily figuring out what version is running.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -7,6 +7,34 @@ const HttpProxyMiddlware = require("http-proxy-middleware");
...
+ *  * 'dirty' if dirty or not on branch 'main'
+ *  * the commit sha1 otherwise
+ */
+function versionInfo() {
+    try {
+        // Grab the branch and trim newline; on failure (for instance not a git repo, git not installed, etc)
+        // just say "unknown"
+        const branch = execSync('git rev-parse --abbrev-ref HEAD', { encoding: 'utf-8'} ).replace(/^\s+|\s+$/g, '');
+        try {
+            if (branch !== "main") { throw Error('not on main'); }
```
Comment (peterpeterparker) [inline webpack.config.js:26]: "not on main" will be reported as "dirty"...just in case it's not expected

**Review by peterpeterparker — APPROVED** [review]
LGTM

File: webpack.config.js (hunk 1 above)
Comment (nmattia) [reply webpack.config.js:26]: Yes, that's expected! I thought about putting the branch name or something but often the branch will later be deleted etc, so I thought it'd be simpler to say "this is not a clean, "main" build". Good catch though!

**Comment by nmattia** [comment]
Just realized there's no git info in the (official) docker build, will need to inject that

**Pull request closed without merging**